server/src/main/java/org/tctalent/server/service/db/impl/LinkPreviewServiceImpl.java

```diff
@@ -0,0 +1,283 @@
+ * along with this program. If not, see https://www.gnu.org/licenses/.
+ */
+
+package org.tctalent.server.service.db.impl;
+
+import static java.lang.Double.parseDouble;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.jsoup.Connection;
+import org.jsoup.HttpStatusException;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.select.Elements;
+import org.springframework.lang.Nullable;
+import org.springframework.stereotype.Service;
+import org.tctalent.server.exception.EntityReferencedException;
+import org.tctalent.server.exception.InvalidRequestException;
+import org.tctalent.server.logging.LogBuilder;
+import org.tctalent.server.model.db.LinkPreview;
+import org.tctalent.server.repository.db.LinkPreviewRepository;
+import org.tctalent.server.service.db.LinkPreviewService;
+
+@Slf4j
+@RequiredArgsConstructor
+@Service
+public class LinkPreviewServiceImpl implements LinkPreviewService {
+
+  private final LinkPreviewRepository linkPreviewRepository;
+
+  @Override
+  public boolean deleteLinkPreview(long linkPreviewId)
+      throws EntityReferencedException, InvalidRequestException {
+    linkPreviewRepository.deleteById(linkPreviewId);
+    return true;
+  }
+
+  @Override
+  public @Nullable LinkPreview buildLinkPreview(String url) throws IOException {
+    LinkPreview linkPreview = new LinkPreview();
+    linkPreview.setUrl(url);
+
+    try {
+      Document doc = Jsoup.connect(url)
+          // Mask Jsoup as browser - evades 403s in some cases.
+          .userAgent("Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.124 Safari/537.36")
+          .header("Accept", "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8")
+          .header("Accept-Encoding", "gzip, deflate, br")
+          .header("Accept-Language", "en-US,en;q=0.9")
+          .header("Connection", "keep-alive")
+          .header("Referer", "http://google.com")
+          .get();
+
+      // A valid domain name is the minimal threshold content for displaying a preview.
+      String domain = getDomain(doc, url);
+      if (domain != null) {
+        linkPreview.setDomain(domain);
+      } else return null;
+
+      linkPreview.setTitle(getTitle(doc));
+      linkPreview.setDescription(getDescription(doc));
+      linkPreview.setImageUrl(getImageUrl(doc));
+      linkPreview.setFaviconUrl(getFaviconUrl(doc));
+
+      return linkPreview;
+
+    } catch (HttpStatusException e) {
+      // If doc can't be retrieved, log the error and return null.
+
+      LogBuilder.builder(log)
+          .action("BuildLinkPreview: Jsoup.connect(url).get()")
+          .message("HTTP error fetching URL " + "(" + url + "): " + e.getStatusCode())
+          .logInfo();
+
+      return null;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An exception has been ignored. The catch-block hides information about the stack trace. Not logging or forwarding the stack trace of an exception can mask unexpected errors and complicated debugging.

**Fix**: Use a catch block to handle the exception and write a log message declaring that the exception happened. 
[Learn more.](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)